bundles/framework/timeseries/WMSAnimator.js

```diff
@@ -0,0 +1,180 @@
+                    this._isLoading = false;
+                    this._resolveWait();
+                }
+            }
+        },
+        onEvent: function (event) {
+            var handler = this.__eventHandlers[event.getName()];
+            if (!handler) {
+                return;
+            }
+            return handler.apply(this, [event]);
+        },
+        /**
+         * @method getTimes
+         * Returns all available time instants that can be shown
+         * @return {String[]} list of available timeseries times
+         */
+        getTimes: function () {
+            var times = this._layer.getAttributes().times;
+            if (!Array.isArray(times)) {
+                var interval = moment.duration(times.interval);
+                var end = moment(times.end);
+                var t = moment(times.start);
+                times = [t.toISOString()];
+                while (t.add(interval) < end) {
+                    times.push(t.toISOString());
+                }
+                times.push(end.toISOString());
+            }
+            return times;
+        },
+        init: function () { },
+        /**
+         * @method getCurrentTime
+         * Returns current selected time instant
+         * @return {String} current time as ISO-string
+         */
+        getCurrentTime: function () {
+            return this._currentTime;
+        },
+        /**
+         * @method getSubsetRange
+         * Returns current selected time range subset
+         * @return {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        getSubsetRange: function () {
+            return this._subsetRange.slice();
+        },
+        /**
+         * @method getSubsetRange
+         * Set current selected time range subset
+         * @param {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        setSubsetRange: function (range) {
+            this._subsetRange = range;
+        },
+        /**
+         * @method requestNewTime
+         * Requests change in current selected time
+         * @param {String} newTime change current time to this value(ISO string)
+         * @param {String} nextTime time value at next animation frame(ISO string). Can be null if not animating
+         * @param {function} doneCallback callback that will be called after new time has been loaded
+         */
+        requestNewTime: function (newTime, nextTime, doneCallback) {
+            var me = this;
+            this._currentTime = newTime;
+            var requestBuilder = this._sandbox.getRequestBuilder('MapModulePlugin.MapLayerUpdateRequest');
+            if (requestBuilder) {
+                this._isLoading = true;
+                this._doneCallback = doneCallback;
+                if (nextTime) {
+                    this._bufferImages(this._mapModule.getLayerTileUrls(this._layer.getId()), nextTime, function (success) {
+                        me._isBuffering = false;
```

Comment:
This isBuffering handling should be moved to _bufferImages() so it keeps track of the status instead of external code that's not actually doing the buffering.

Reply:
Good catch. Actually I would solve the problem the other way around: move `this._isBuffering = true;` from `_bufferImages` to `requestNewTime`. In this way _bufferImages would not mutate the WMSAnimator instance state, and therefore it could be called from many call sites independently if needed.